bftengine/include/bftengine/DbCheckpointManager.hpp

```diff
@@ -196,6 +197,8 @@ class DbCheckpointManager {
   mutable std::mutex lock_;
   mutable std::mutex lockDbCheckPtFuture_;
   std::mutex lockLastDbCheckpointDesc_;
+  std::condition_variable shutdownCond_;  // used to wake up monitor thread periodically or when destructing an obj
+  std::mutex lockShutdown_;          // used for the above cond var. doesn't really coordinate anything between threads.
```

Comment:
I think that if you ended with cond, you should end with lock. Also, the comment of this line is completely redundent in my opinion for anyone with basic c++ knowledge.
`std::mutex shutdownLock_;`